### PR TITLE
MNT: Use noarch python {{ python_min }} variable

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Use `python {{ python_min }}` syntax for the python requirements for noarch python recipes.
   - c.f. https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python
   - c.f. https://github.com/conda-forge/uproot-feedstock/pull/160#issuecomment-2463401308
* Add `recipe/conda_build_config.yaml` to override the global `python_min` with uproot's
  `python_min` of `3.8`.
* Bump build number.
* MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.44.2, and conda-forge-pinning 2024.11.07.16.38.57

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

```

* Use 'python {{ python_min }}' syntax for the python requirements for noarch
  python recipes.
   - c.f. https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python
* Add recipe/conda_build_config.yaml to override the global python_min with uproot's
  python_min of 3.8.
* Bump build number.
* MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.44.2, and conda-forge-pinning 2024.11.07.16.38.57
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -6,3 +6,11 @@ channel_targets:
...
 docker_image:
 - quay.io/condaforge/linux-anvil-cos7-x86_64
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.12.* *_cpython
+python_min:
+- '3.9'
```
Comment (matthewfeickert) [inline .ci_support/linux_64_.yaml:16]: > You may need to override the python_min variable in the `conda_build_config.yaml`/`variants.yaml` if the package requires a newer Python version than the currently supported minimum version on conda-forge.

Okay, so I now need to figure out how to override `python_min` as to be '3.8'.

Comment (matthewfeickert) [reply .ci_support/linux_64_.yaml:16]: @beckermr @chrisburr @jakirkham This isn't very clear from https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#general-pinning-examples (as there's nothing on this in https://conda-forge.org/docs/). Do we need to create a `recipe/conda_build_config.yaml` that is _just_

```yaml
python_min:
- '3.8'
```

or do we need to include all possible Python information in it, essentially duplicating all information in `.ci_support/linux_64_.yaml`

```yaml
pin_run_as_build:
  python:
    min_pin: x.x
    max_pin: x.x
python:
- 3.12.* *_cpython
python_min:
- '3.8'
```

?

Comment (matthewfeickert) [reply .ci_support/linux_64_.yaml:16]: cc @dopplershift given his participation (and shared concerns on supporting older Pythons) in https://github.com/conda-forge/conda-forge.github.io/issues/2210. Ryan, maybe you have insight here as to how this works?

Comment (beckermr) [reply .ci_support/linux_64_.yaml:16]: Hopefully that works. If not, we can try other things.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_.yaml (hunk 1 above)
Comment (matthewfeickert) [reply .ci_support/linux_64_.yaml]: Thanks very much @beckermr. That indeed does work. :+1: 

If I find time this evening I will open and Issue and PR to make things a bit more explicit for people who have only been doing this for a few years like me.

Comment (jakirkham) [reply .ci_support/linux_64_.yaml]: Improvements to the docs would certainly be welcome 🙂

Writing good docs seems to be one of the hardest problems in open source. So we welcome all the help we can get 😅

Comment (matthewfeickert) [reply .ci_support/linux_64_.yaml]: It's hard, and also the conda-forge team has an nearly impossible task of covering a _huge_ landscape of complicated infrastructure (that still amazingly "just works"!).

Given that I'm trying to unbreak some Scikit-HEP related builts that might not happen tonight, but my goal is to open up a PR in the next 24 hours to iterate with you all on. :+1:

Comment (matthewfeickert) [reply .ci_support/linux_64_.yaml]: Okay, so closer to 96 hours, but: https://github.com/conda-forge/conda-forge.github.io/pull/2370

**Pull request merged**